### PR TITLE
Add suport for distributing work based on a hashed term

### PR DESCRIPTION
**Pull request by JoshRagem**
`erlang:phash2(HashTerm, Wpool_Size)` is used to calculate an index to target a worker in the pool.

example of use:

```
wpool:cast(?MODULE,{work_harder, Tasks}, {hash_worker, UserId})
%% or
wpool:call(?MODULE,{work_harder, Tasks}, {hash_worker, UserId})
```

### CODE REVIEW

File: src/wpool_pool.erl [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 %% API
 -export([start_link/2, create_table/0]).
--export([best_worker/1, random_worker/1, next_worker/1, available_worker/2]).
+-export([best_worker/1, random_worker/1, next_worker/1, available_worker/2, hash_worker/2]).
```
Comment (gadgetci) [inline src/wpool_pool.erl]: According to **Elvis**:
> Line 24 is too long: -export([best_worker/1, random_worker/1, next_worker/1, available_worker/2, hash_worker/2])..

**Comment by elbrujohalcon** [comment]
@JoshRagem can you add tests for this new strategy, please?

**Comment by JoshRagem** [comment]
I will try to add tests soon if you all are happy with this strategy

**Comment by elbrujohalcon** [comment]
@JoshRagem the strategy looks a little bit different than the others in the sense that it's somewhat more _deterministic_ or predictable, but the implementation seems sound

**Comment by JoshRagem** [comment]
@elbrujohalcon I added a test based on what I found in `wpool_pool_SUITE.erl`

**Comment by elbrujohalcon** [comment]
awesome, @JoshRagem !! Thank you very much for this!

**Pull request merged**